tests/utils/streams/test_kafka.py

```diff
@@ -17,268 +18,147 @@
-        assert (
-            wait(
-                [producer.produce(topic, KafkaPayload(None, value)) for i in range(2)],
-                timeout=5.0,
-            ).not_done
-            == set()
+    def get_producer(self) -> KafkaProducer[int]:
+        return KafkaProducer(self.configuration, self.codec)
+
+    def test_auto_offset_reset_earliest(self) -> None:
+        with self.get_topic() as topic:
+            with closing(self.get_producer()) as producer:
+                producer.produce(topic, 0).result(5.0)
+
+            with closing(
+                self.get_consumer("test", auto_offset_reset="earliest")
+            ) as consumer:
+                consumer.subscribe([topic])
+
+                message = consumer.poll(10.0)
+                assert isinstance(message, Message)
+                assert message.offset == 0
+
+    def test_auto_offset_reset_latest(self) -> None:
+        with self.get_topic() as topic:
+            with closing(self.get_producer()) as producer:
+                producer.produce(topic, 0).result(5.0)
+
+            with closing(
+                self.get_consumer("test", auto_offset_reset="latest")
+            ) as consumer:
+                consumer.subscribe([topic])
+
+                try:
+                    consumer.poll(10.0)  # XXX: getting the subcription is slow
+                except EndOfPartition as error:
+                    assert error.partition == Partition(topic, 0)
+                    assert error.offset == 1
+                else:
+                    raise AssertionError("expected EndOfPartition error")
+
+    def test_auto_offset_reset_error(self) -> None:
+        with self.get_topic() as topic:
+            with closing(self.get_producer()) as producer:
+                producer.produce(topic, 0).result(5.0)
+
+            with closing(
+                self.get_consumer("test", auto_offset_reset="error")
+            ) as consumer:
+                consumer.subscribe([topic])
+
+                with pytest.raises(ConsumerError):
+                    consumer.poll(10.0)  # XXX: getting the subcription is slow
+
+    def test_commit_log_consumer(self) -> None:
+        # XXX: This would be better as an integration test (or at least a test
+        # against an abstract Producer interface) instead of against a test against
+        # a mock.
+        commit_log_producer = FakeConfluentKafkaProducer()
+
+        consumer: KafkaConsumer[int] = KafkaConsumerWithCommitLog(
+            {
+                **self.configuration,
+                "auto.offset.reset": "earliest",
+                "enable.auto.commit": "false",
+                "enable.auto.offset.store": "false",
+                "enable.partition.eof": "true",
+                "group.id": "test",
+                "session.timeout.ms": 10000,
+            },
+            codec=self.codec,
+            producer=commit_log_producer,
+            commit_log_topic=Topic("commit-log"),
         )
```

Comment:
This (the commit log stuff) is very close to being a wrapper around a `Consumer` and `Producer` instance instead of being Kafka-specific.